File: src/CazaTalentos/zbasket_03.jl
using Random

Random.seed!(102191)

# calcula cuantos encestes logra una jugadora con indice de enceste prob
# que hace qyt tiros libres

function ftirar(prob, qty)
  return  sum(rand() < prob for i in 1:qty)
end


# defino los jugadores
taurasi = [0.7]
peloton = Vector((501:599) / 1000)
jugadoras = append!(taurasi, peloton)

# veo que tiene el vector
jugadoras




for i = 1:10
  vaciertos = ftirar.(jugadoras, 10)  # 10 tiros libres cada jugadora
  mejor = findmax(vaciertos)[2]
  aciertos_torneo = vaciertos[ mejor ]

  aciertos_segunda = ftirar.(jugadoras[ mejor ], 10)

  println(aciertos_torneo, "\t", aciertos_segunda)
end

File: src/CazaTalentos/zbasket_03.py
import  numpy as np

np.random.seed(102191)

# calcula cuantos encestes logra una jugadora con indice de enceste prob
# que hace qyt tiros libres

def ftirar(prob, qty):
  return sum(np.random.rand(qty) < prob)



# defino los jugadores
taurasi = 0.7
peloton = np.array(range(501, 600)) / 1000
jugadoras = np.append(taurasi, peloton)

# veo que tiene el vector
jugadoras

# vectorizo la funcion  ftirar
vec_ftirar = np.vectorize(ftirar)

for i in range(10):
  vaciertos = vec_ftirar(jugadoras, 10) # 10 tiros libres cada jugadora
  mejor = np.argmax(vaciertos)
  aciertos_torneo = vaciertos[mejor]
  aciertos_segunda = vec_ftirar(jugadoras[mejor], 10)
  print(aciertos_torneo, "\t", aciertos_segunda)


